hooks/matrix.ts

```diff
@@ -0,0 +1,295 @@
+      link: generateLink(config, null, y),
+    };
+
+    const gridRow: any = [];
+    xData.forEach((x: MatrixDatum, col: number) => {
+      // Parse Columns
+      const xParse = {
+        id: x.id,
+        title: x.title,
+        link: generateLink(config, x),
+      };
+
+      let cellLink = '';
+      const cellColors: Array<number> = [];
+      if (xyzMap[x.id] && xyzMap[x.id][y.id]) {
+        const currZ = xyzMap[x.id][y.id];
+        cellLink = generateLink(config, x, y);
+        currZ.forEach((zTitle: string) => {
+          cellColors.push(colorMap[zTitle]);
+        });
+      }
+
+      const cellId = `${x.id} + ${y.id}`;
+      const cellTitle = `${x.title} + ${y.title}`;
+
+      gridRow.push({
+        row: yParse,
+        column: xParse,
+        id: cellId,
+        title: cellTitle,
+        link: cellLink,
+        colors: cellColors,
+      });
+    });
+
+    // Add empty column at last of each row as a margin
+    gridRow.push({
+      row: { id: y.id, title: '', link: '' },
+      column: { id: xData.length, title: '', link: '' },
+      id: `${y.id}-${row}-${xData.length}`,
+      title: '',
+      link: '',
+      colors: [],
+    });
+    gridData.push(gridRow);
+  });
+
+  // Add empty row as a margin
+  const emptyRow = xData.map((x: any, col: number) => ({
+    row: { id: yData.length, title: '', link: '' },
+    column: { id: x.id, title: '', link: '' },
+    id: `${x.id}-${col}-${yData.length}`,
+    title: '',
+    link: '',
+    colors: [],
+  }));
+  // add last corner cell
+  emptyRow.push({
+    row: { id: yData.length, title: '', link: '' },
+    column: { id: xData.length, title: '', link: '' },
+    id: `${xData.length}-${yData.length}`,
+    title: '',
+    link: '',
+    colors: [],
+  });
+  gridData.push(emptyRow);
+
+  // Create the options and datamap used for the matrix search feature
+  const options: Array<any> = [];
+  const gridDataMap: any = {};
+  yData.forEach((y: MatrixDatum, row: number) => {
+    options.push({ value: y.id, label: y.title });
+    gridDataMap[y.title.toLowerCase()] = { type: 'row', index: row };
+  });
+  xData.forEach((x: MatrixDatum, col: number) => {
+    options.push({ value: x.id, label: x.title });
+    gridDataMap[x.title.toLowerCase()] = { type: 'col', index: col };
+  });
+
+  const parsedData: any = {
+    gridData,
+    legendData,
+    gridDataMap,
+    options,
+  };
+
+  return parsedData;
+};
+
+// Selectable color options for the matrix
+const colorOptions = [
+  { value: 'default', label: 'Default' },
+  { value: 'parula', label: 'Parula' },
+  { value: 'viridis', label: 'Viridis' },
+];
+
+/**
+ * Hook function to use matrix data given a config object
+ *
+ * @param matrixConfigs
+ * @returns all data to be used by matrix visualization
+ */
+export const useMatrixData = (matrixConfigs: any) => {
+  const { dispatchError, errors } = useError();
+  const [styles, setStyles] = useState<any>(null); // styles from the config object
```

Comment:
Move the comments about the `useState` line to make it easier to read for lengthier lines.

Reply:
I think typically state comments are not that fleshed out. I typically don't see any more comments than this for useState. 

More complicated state can be built into a reducer and documented on that instead.
I have moved it to the preferred style though.